app/core/speaker_diarization.py

```
@@ -0,0 +1,312 @@
+        response = requests.get(gcs_url, timeout=600)  # 10분 (600초)
+        response.raise_for_status()
+        
+        # 파일 크기 체크 (500MB 제한)
+        content_length = len(response.content)
+        max_size = 500 * 1024 * 1024  # 500MB
+        
+        if content_length > max_size:
+            raise HTTPException(
+                status_code=400, 
+                detail=f"오디오 파일이 너무 큽니다: {content_length / (1024*1024):.1f}MB (최대 500MB)"
+            )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)